sklearn/preprocessing/data.py

```diff
@@ -229,11 +233,17 @@ class MinMaxScaler(BaseEstimator, TransformerMixin):
 
     scale_ : ndarray, shape (n_features,)
         Per feature relative scaling of the data.
+
+
+
     """
 
-    def __init__(self, feature_range=(0, 1), copy=True):
+    def __init__(self, feature_range=(0, 1), copy=True, selected="all"):
         self.feature_range = feature_range
         self.copy = copy
+        self.selected = selected
+        self.scaled_features_ = None if selected == "all" else selected
```

Comment:
I suppose the only reservation I have with this is, if I move this check...

```
 if selected != "all":
      if any(isinstance(index, str) for index in selected):
            raise ValueError("Selected features must be ints, not strings")
```

into `fit`, then I miss the opportunity to raise the error in the constructor. We could be catching this error sooner in a user's workflow than later.

Should I still move it to `fit`?